kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java

```diff
@@ -46,4 +48,32 @@ private CompletableFutures() {
     future.completeExceptionally(exception);
     return future;
   }
+
+  /**
+   * Returns a new {@link CompletableFuture} that is completed when {@code future} is complete,
+   * catching the given {@link Throwable exceptionClass}.
+   *
+   * <p>If {@code future} completes normally, then the returned future completes normally with the
+   * same value. If {@code future} completes exceptionally with an {@code exceptionClass}, the
+   * returned completes with the result of the {@code handler}. Otherwise, the returned future will
+   * complete exceptionally with the same exception as {@code future}.
+   */
+  public static <T, E extends Throwable> CompletableFuture<T> catchingCompose(
+      CompletableFuture<T> future,
+      Class<E> exceptionClass,
+      Function<? super E, ? extends CompletableFuture<T>> handler) {
+    return future.handle(
+        (value, error) -> {
+          if (error == null) {
+            return CompletableFuture.completedFuture(value);
+          } else if (exceptionClass.isInstance(error.getCause())) {
+            return handler.apply(exceptionClass.cast(error.getCause()));
+          } else if (error instanceof CompletionException) {
+            throw (CompletionException) error;
+          } else {
+            throw new AssertionError(error); // impossible
```

Comment:
nit: replace with "Should not happen"; if it's truly impossible this else need not exist.

Reply:
Done.